iceoryx_examples/user_header/README.md

```diff
@@ -0,0 +1,337 @@
+```cpp
+#include "user_header_and_payload_types.hpp"
+
+#include "iceoryx_posh/popo/publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+```
+
+The `main` function contains a loop which periodically sends a Fibonacci number.
+In order to break this loop, a signal handler is used which sets an atomic boolean.
+<!-- [geoffrey] [iceoryx_examples/user_header/publisher_cxx_api.cpp] [signal handling] -->
+```cpp
+std::atomic<bool> killswitch{false};
+
+static void sigHandler(int sig IOX_MAYBE_UNUSED)
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+```
+
+The signal handler is registered in the `main` function.
+<!-- [geoffrey] [iceoryx_examples/user_header/publisher_cxx_api.cpp] [register sigHandler] -->
+```cpp
+auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+```
+
+Next, the iceoryx runtime is initialized. With this call, the application will be registered at `RouDi`,
+the routing and discovery daemon.
+<!-- [geoffrey] [iceoryx_examples/user_header/publisher_cxx_api.cpp] [initialize runtime] -->
+```cpp
+constexpr char APP_NAME[] = "iox-cpp-user-header-publisher";
+iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+```
+
+Now, we create the publisher. Unlike the other examples, this uses the second template parameter to define the user-header.
+This is the only change compared to the other examples with the typed C++ API.
+<!-- [geoffrey] [iceoryx_examples/user_header/publisher_cxx_api.cpp] [create publisher] -->
+```cpp
+// for the 1.0 release, the Publisher alias for the PublisherImpl does not have the second parameter for the Header,
+// therefore the PublisherImpl must be used directly
+iox::popo::PublisherImpl<Data, Header> publisher({"Example", "User-Header", "Timestamp"});
+```
+
+In the main loop, a Fibonacci sequence is created and every second a number is published.
+The Fibonacci number is passed to the `loan` method which construct the `Data` object if the loaning was successful.
+This example uses the functional approach with `and_then` and `or_else`. Please have a look at the `icedelivery` example
+if you prefer a more traditional approach.
+<!-- [geoffrey] [iceoryx_examples/user_header/publisher_cxx_api.cpp] [[send samples in a loop] [loan sample]] -->
+```cpp
+uint64_t timestamp = 0;
+uint64_t fibonacciLast = 0;
+uint64_t fibonacciCurrent = 1;
+while (!killswitch)
+{
+    auto fibonacciNext = fibonacciCurrent + fibonacciLast;
+    fibonacciLast = fibonacciCurrent;
+    fibonacciCurrent = fibonacciNext;
+
+    publisher.loan(Data{fibonacciCurrent})
+        .and_then([&](auto& sample) {
+            // ...
+        })
+        .or_else([&](auto& error) {
+            // ...
+        });
+
+    constexpr uint64_t SLEEP_TIME{1000U};
```

Comment:
SLEEP_TIME_MS? Or even better use `units::Duration` or `chrono::milliseconds` for the variable.

Reply:
I use that constant to increment the timestamp in the header, therefore I would use `SLEEP_TIME_MS`